### PR TITLE
added cors

### PR DESCRIPTION
**Pull request by nunodamaso**
Added required headers to allow cross-origins requests (a browser requesting the API).

### CODE REVIEW

File: server/impl/DealApiImpl.cpp [hunk 1]
```diff
@@ -69,7 +69,7 @@ void DealApiImpl::get_deal(const Pistache::Optional<std::string> &seed, const Pi
       ww_deal_to_uuid(deal_uid, uuid);
       deals.push_back(std::string(uuid));
    }
-
+   response.headers().add<Http::Header::AccessControlAllowOrigin>("*"); // ACCESS CONTROL ALLOW ORIGIN (CORS)
```
Comment (suokko) [inline server/impl/DealApiImpl.cpp]: Moving the comment from the end line to the previous line would help reduce line length.

**Comment by tameware** [comment]
No compatible CORS library for C++? That's a shame - Nuno's solution was super-easy with Flask in Python for DDS. But doing it by hand is not a problem.

**Comment by nunodamaso** [comment]
Hot it, I'll hotfix it later today

**Comment by nunodamaso** [comment]
I didn't find any, from what I understood it had to be done manually

**Comment by suokko** [comment]
C++ had a long period when it wasn't really used for web development. Libraries are either recent or abandoned before C++11 standard. https://www.webtoolkit.eu/wt might have something but I haven't looked into in depth.

**Comment by nunodamaso** [comment]
Moved comment

File: server/impl/DealApiImpl.cpp [hunk 2]
```diff
@@ -69,7 +69,8 @@ void DealApiImpl::get_deal(const Pistache::Optional<std::string> &seed, const Pi
       ww_deal_to_uuid(deal_uid, uuid);
       deals.push_back(std::string(uuid));
    }
-
+//    ACCESS CONTROL ALLOW ORIGIN (CORS)
```
Comment (suokko) [inline server/impl/DealApiImpl.cpp]: Comments are usually intended to same depth as following code line.

**Review by suokko — COMMENTED** [review]
It would be a good idea to use `git commit --amend` or `git rebase -i` when fixing commits.

This way development history becomes easier to analyze. The analyzing history can be helpful sometimes when trying to understand reason behind some code.

**Comment by nunodamaso** [comment]
I would argue that rebasing *destroys* the commit history for further
analyses, while always merging without fast-forwarding *keeps truth* to
development. If it's easier or not to analyse is subjective :)

suokko <notifications@github.com> escreveu no dia quarta, 15/07/2020 à(s)
11:26:

**Comment by nunodamaso** [comment]
Also, I'm finding I can't write a single line of code without a linter
anymore! :D Ill fix the tab advance

Nuno Dâmaso <nuno.s.damaso@gmail.com> escreveu no dia quarta, 15/07/2020
à(s) 11:42:

**Comment by suokko** [comment]
On Wed, Jul 15, 2020 at 2:43 PM nunodamaso <notifications@github.com> wrote:

> I would argue that rebasing *destroys* the commit history for further
> analyses, while always merging without fast-forwarding *keeps truth* to
> development. If it's easier or not to analyse is subjective :)
>
>
I think you misunderstood my meaning. I was talking about having only one
commit for related changes. In this case all changes are related, history
would be easier to read if you have only one commit. To turn changes into
one commit `git commit --amend` is a useful tool.

In case a change set consists of multiple related commits then `git rebase
-i` is a helpful tool when fixing earlier commits in the branch. The
argument switches to interactive rebase mode. Interactive mode allows
editing changes, rewording commit messages, reordering commits and merging
changes together from two commits.

Development history before pull requests are merged rarely is useful
information. Changes before merge are often incremental improvements to the
change which would be later easier to understand as one commit.

**Comment by nunodamaso** [comment]
Indeed I missed the -i and the point. Thanks that looks super useful.

**Review by woefulwabbit — APPROVED** [review]

**Comment by suokko** [comment]
I decided to fix the style issues. Then I noticed template keyword is missing from a template method call.

**Pull request merged**